### PR TITLE
Update transfers inline with latest spec

### PR DESCRIPTION
**Pull request by br648**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR seeks to update transfers to be inline with the latest spec which covers additional parameters: https://github.com/google/transit/blob/master/gtfs/spec/en/reference.md#transferstxt

The validation of transfers is to be managed by the MobilityData Validator and as such validation is outside the scope of this PR.

Jira ticket: https://ibisolutions.atlassian.net/jira/software/c/projects/DT/issues/DT-209

### CODE REVIEW

File: src/main/java/com/conveyal/gtfs/model/Transfer.java [hunk 1]
```diff
@@ -54,24 +57,27 @@ protected boolean isRequired() {
...
+            getRefField("to_stop_id", false, feed.stops);
             getRefField("from_route_id", false, feed.routes);
             getRefField("to_route_id", false, feed.routes);
             getRefField("from_trip_id", false, feed.trips);
             getRefField("to_trip_id", false, feed.trips);
 
             tr.feed = feed;
-            feed.transfers.put(Long.toString(row), tr);
+            feed.transfers.put(
+                createId(tr.from_stop_id, tr.to_stop_id, tr.from_trip_id, tr.to_trip_id, tr.from_route_id, tr.to_route_id),
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/model/Transfer.java]: Replace with `tr.getId()`.

File: src/main/java/com/conveyal/gtfs/model/Transfer.java [hunk 2]
```diff
@@ -99,7 +118,20 @@ protected void writeOneRow(Transfer t) throws IOException {
...
+     */
+    private static String createId(
+        String fromStopId,
+        String toStopId,
+        String fromTripId,
+        String toTripId,
+        String fromRouteId,
+        String toRouteId
+    ) {
+        return String.format("%s_%s_%s_%s_%s_%s", fromStopId, toStopId, fromTripId, toTripId, fromRouteId, toRouteId);
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/model/Transfer.java:131]: I suppose this supports null values on some of the optional fields.

File: src/main/java/com/conveyal/gtfs/model/Transfer.java [hunk 3]
```diff
@@ -99,7 +118,20 @@ protected void writeOneRow(Transfer t) throws IOException {
...
     }
+
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/gtfs/model/Transfer.java]: Revert this blank line addition.

File: src/test/resources/fake-agency/transfers.txt [hunk 4]
```diff
@@ -1 +1,2 @@
-from_stop_id,to_stop_id,transfer_type,min_transfer_time
+from_stop_id,to_stop_id,from_trip_id,to_trip_id,from_route_id,to_route_id,transfer_type,min_transfer_time
```
Comment (binh-dam-ibigroup) [inline src/test/resources/fake-agency/transfers.txt:1]: Should there be a case for when not all columns are populated? (e.g. for a more generic transfer policy)

**Review by binh-dam-ibigroup — APPROVED** [review]
Looks solid. I'll let the other reviewer decide if more test entries in transfers.txt should be included.

File: src/main/java/com/conveyal/gtfs/model/Transfer.java (hunk 1 above)
Comment (br648) [reply src/main/java/com/conveyal/gtfs/model/Transfer.java]: Good spot!

File: src/main/java/com/conveyal/gtfs/model/Transfer.java (hunk 2 above)
Comment (br648) [reply src/main/java/com/conveyal/gtfs/model/Transfer.java:131]: It produces for example: `4u6g_123_null_null_null_null`, which will still be unique with other rows that have not defined optional fields providing the from and to stop ids are unique (which would flag a duplicate error if not).

File: src/test/resources/fake-agency/transfers.txt (hunk 4 above)
Comment (br648) [reply src/test/resources/fake-agency/transfers.txt:1]: Yes, I have added.

**Review by philip-cline — APPROVED** [review]
Looks really good, thanks for the update. I think the tests are fine as is.

**Pull request merged**